src/checkqueue.h

```diff
@@ -167,15 +167,20 @@ class CCheckQueue
     //! Add a batch of checks to the queue
     void Add(std::vector<T>& vChecks)
     {
-        LOCK(m_mutex);
-        for (T& check : vChecks) {
-            queue.push_back(T());
-            check.swap(queue.back());
+        unsigned int num_of_new_checks;
```

Comment:
`unsigned int` is typically smaller than the size of a vector. Use `size_t` or `std::vector<T>::size_type` perhaps?

Reply:
I've chosen `unsinged int` due to https://github.com/bitcoin/bitcoin/blob/5adc5c02800f00d1e6e8812a2b0559b1800e82e9/src/checkqueue.h#L60

Maybe use `size_t` for `nTodo` as well?

Reply:
Ah, I see, that makes sense. I think changing to size_t everywhere can be done in a separate PR, if desires.

Reply:
the max number of checks is IIRC something slightly above 100k